src/datasources/email/email.datasource.spec.ts

```diff
@@ -440,7 +466,255 @@ describe('Email Datasource Tests', () => {
+
+  it('subscribes to category', async () => {
+    const chainId = faker.number.int({ max: DB_CHAIN_ID_MAX_VALUE }).toString();
+    const safeAddress = faker.finance.ethereumAddress();
+    const emailAddress = new EmailAddress(faker.internet.email());
+    const account = faker.finance.ethereumAddress();
+    const code = faker.string.numeric();
+    const codeGenerationDate = faker.date.recent();
+    const unsubscriptionToken = faker.string.uuid();
+    const subscription = {
+      key: faker.word.sample(),
+      name: faker.word.words(2),
+    };
+    await sql`INSERT INTO emails.subscriptions (key, name)
+              VALUES (${subscription.key}, ${subscription.name})`;
+    await target.saveEmail({
+      chainId,
+      safeAddress,
+      emailAddress,
+      account,
+      code,
+      codeGenerationDate,
+      unsubscriptionToken,
+    });
+
+    await target.subscribe({
+      chainId,
+      safeAddress,
+      account,
+      categoryKey: subscription.key,
+    });
+
+    const subscriptions = await target.getSubscriptions({
+      chainId,
+      safeAddress,
+      account,
+    });
+    expect(subscriptions).toContainEqual(subscription);
+  });
+
+  it('unsubscribes to a category successfully', async () => {
+    const chainId = faker.number.int({ max: DB_CHAIN_ID_MAX_VALUE }).toString();
+    const safeAddress = faker.finance.ethereumAddress();
+    const emailAddress = new EmailAddress(faker.internet.email());
+    const account = faker.finance.ethereumAddress();
+    const code = faker.string.numeric();
+    const codeGenerationDate = faker.date.recent();
+    const unsubscriptionToken = faker.string.uuid();
+    const subscriptions = [
+      {
+        key: faker.word.sample(),
+        name: faker.word.words(2),
+      },
+      {
+        key: faker.word.sample(),
+        name: faker.word.words(2),
+      },
+    ];
+    await sql`INSERT INTO emails.subscriptions (key, name)
+              VALUES (${subscriptions[0].key}, ${subscriptions[0].name}),
+                     (${subscriptions[1].key}, ${subscriptions[1].name})`;
+    await target.saveEmail({
+      chainId,
+      safeAddress,
+      emailAddress,
+      account,
+      code,
+      codeGenerationDate,
+      unsubscriptionToken,
+    });
+    // Subscribe to two categories
+    await target.subscribe({
+      chainId,
+      safeAddress,
+      account,
+      categoryKey: subscriptions[0].key,
+    });
+    await target.subscribe({
+      chainId,
+      safeAddress,
+      account,
+      categoryKey: subscriptions[1].key,
+    });
+
+    // Unsubscribe from one category
+    const result = await target.unsubscribe({
+      categoryKey: subscriptions[0].key,
+      token: unsubscriptionToken,
+    });
+
+    const currentSubscriptions = await target.getSubscriptions({
+      chainId,
+      safeAddress,
+      account,
+    });
+    expect(result).toContainEqual(subscriptions[0]);
+    expect(currentSubscriptions).not.toContainEqual(subscriptions[0]);
```

Comment:
I'd suggest we also add a check for the subscription of `subscriptions[1]` here to be on the safe side.

Reply:
This is kind of covered in `expect(result).toContainEqual(subscriptions[0])` since it should return the subscriptions removed but I will add the assertion which I think is also valid!

Reply:
Addressed in [040d7d6](https://github.com/safe-global/safe-client-gateway/pull/1028/commits/040d7d6378f9fe26daf364d342d3a9481c361b8a)